application/associations.py

```diff
@@ -263,7 +263,10 @@ def edit_file(self, selection):
 			self._application.create_terminal_tab(active_object._notebook, options)
 
 		else:
-			subprocess.Popen(split_command, cwd=os.path.dirname(selection[0]))
+			cwd = None
```

Comment:
For non-local selection[0] (e.g. any GIO provided source starting with protocol) the existing line fails miserably. The right thing might be a function on the provider to check if you can cwd() on location supported by it. But non of the callers of this function have a provider so it would have to be passed done from further up the stack. Would you want that? Might be a bigger rework...

On the other hand the provider might be able to provide abstraction for mapping logic in line 212 as well.

Let me know what you reckon!